cpp/modmesh/toggle/RadixTree.hpp

```diff
@@ -179,5 +201,146 @@ class RadixTree
     key_type m_unique_id = 0;
 }; /* end class RadixTree */
 
+// The profiling result of the caller
+struct CallerProfile
+{
+    void start_stopwatch()
+    {
+        start_time = std::chrono::high_resolution_clock::now();
+        is_running = true;
+    }
+
+    void stop_stopwatch()
+    {
+        auto end_time = std::chrono::high_resolution_clock::now();
+        auto elapsed_time = std::chrono::duration_cast<std::chrono::nanoseconds>(end_time - start_time);
+        total_time += elapsed_time;
+        call_count++;
+    }
+
+    std::chrono::high_resolution_clock::time_point start_time;
+    std::function<void()> cancel_callback;
+    std::string caller_name;
+    std::chrono::nanoseconds total_time = std::chrono::nanoseconds(0); /// use nanoseconds to have higher precision
+    int call_count = 0;
+    bool is_running = false;
+}; /* end struct CallerProfile */
+
+namespace detail
+{
+class CallProfilerTest; // for gtest
+}
```

Comment:
You forgot the end marking for the namespace

Reply:
@yungyuc  sorry, fixed.